### ISSUE TITLE
Giphy embed not working as expected

### ISSUE THREAD

**Issue by jasonyingling** [WordPress/gutenberg#9003]
**Describe the bug**
When pasting a Giphy page URL or gif link without selecting the Embed block the gif isn't auto-embedded. If you insert the same link directly into an Embed block it will work. Subsequent pasting of the same link itself without selecting the embed block works. 

When publishing gifs appear on page.

When returning to edit the page, Giphy embed links appear broken.

**To Reproduce**
1. Create a new Post
2. Paste a Giphy page URL or direct .gif link without selecting a block
3. Gif appears as broken image
4. Add a new Embed block with Giphy URL or .gif link
5. Paste same Giphy page URL or direct .gif link without selecting a block
6. Publish post
7. View post
8. Return to the edit post screen
9. Gifs appear as broken images

**Expected behavior**
I expect the Giphy page URL to automatically embed the gif without selecting the embed block the first time.

**Screenshots**
**Screencast:** https://monosnap.com/direct/sOmM4EkqDjLK9Nmyicj8kcTJoclVF6
Link used: https://giphy.com/gifs/movie-bruce-willis-die-hard-fY4AWE5NkNRhS

Initial Giphy link inserted without selecting a block
![Initial link insert](https://monosnap.com/image/g9aNJphgLmTFlOjXw9moqa6lyv7hCL)

Same link inserted into Embed block
![link in embed block](https://monosnap.com/image/fMGLRvnhbprd5tLwqQTGs5JEJ45bXI)

When returning to post edit screen after publishing
![after publishing](https://monosnap.com/image/uY8oEvFe2aAmEf9iL44H2yMhsP6R8c)

**Desktop (please complete the following information):**
 - OS: macOS 10.13.6
 - Browser: Chrome 67.0.3396.99
 - Browser: Safari 11.1.2
 - Browser: Firefox 61.0.1

**Additional context**
- Gutenberg 3.5.0

**Comment by brandonpayton** [comment]
Tested and confirmed.

**Comment by webdevdaemon** [comment]
I'm looking for a good 2nd contribution, but am new to the Gutenberg codebase, I am interested in tackling this issue, but I'm not quite sure where to start.

I can figure it out on my own, I just need an idea of where I should start?

Thanks in advance to the kindly gentleperson & scholar that helps me out on this!

- cm

**Comment by neddonovan13** [comment]
Any updates on this one?

**Comment by gziolo** [comment]
There was some initial work done by @dsawardekar in #9344. This issue is still not resolved and it needs a developer who would continue explorations. In the referenced PR there are some useful comments from @notnownikki which should help drive implementation.

**Comment by paaljoachim** [comment]
Testing with WordPress 5.7 beta 3.
Gutenberg plugin version 10.
Twenty Twenty One.


> Create a new Post
> Paste a Giphy page URL or direct .gif link without selecting a block
> Gif appears as broken image

---> I pasted a Giphy page URL and dragged and dropped a .gi file directly into Gutenberg.
Both worked fine.

> Add a new Embed block with Giphy URL or .gif link
> Paste same Giphy page URL or direct .gif link without selecting a block
> Publish post
> View post
> Return to the edit post screen
> Gifs appear as broken images

---> I added an embed block.
Pasted the Giphy URL into it. https://giphy.com/stickers/sad-sadboy-clow-ydmhzlWXnOqwRZ2LUU
It worked in the backend and frontend.
Going to frontend and back to backend the Gifs are not broken.

I would say that this issue has been fixed, and I will go ahead and close it.
